### ISSUE TITLE
Error with MariaDB Database

### ISSUE THREAD

**Issue by Sergix7720** [lucko/LuckPerms#2706]
Recently we updated MariaDB to the version (10.5.7) and Luckperms can't connect anymore with the database, we get this error.

```[17:41:19 ERROR] [LuckPerms]: Exception occurred whilst enabling SQL messaging service
java.lang.RuntimeException: java.sql.SQLException: Malformed communication packet
        at me.lucko.luckperms.common.messaging.sql.SqlMessenger.init(SqlMessenger.java:57) ~[?:?]
        at me.lucko.luckperms.common.messaging.MessagingFactory$SqlMessengerProvider.obtain(MessagingFactory.java:154) ~[?:?]
        at me.lucko.luckperms.common.messaging.LuckPermsMessagingService.<init>(LuckPermsMessagingService.java:73) ~[?:?]
        at me.lucko.luckperms.common.messaging.MessagingFactory.getServiceFor(MessagingFactory.java:106) ~[?:?]
        at me.lucko.luckperms.bungee.messaging.BungeeMessagingFactory.getServiceFor(BungeeMessagingFactory.java:64) ~[?:?]
        at me.lucko.luckperms.common.messaging.MessagingFactory.getInstance(MessagingFactory.java:84) ~[?:?]
        at me.lucko.luckperms.common.plugin.AbstractLuckPermsPlugin.enable(AbstractLuckPermsPlugin.java:149) ~[?:?]
        at me.lucko.luckperms.bungee.LPBungeeBootstrap.onEnable(LPBungeeBootstrap.java:153) ~[?:?]
        at net.md_5.bungee.api.plugin.PluginManager.enablePlugins(PluginManager.java:300) ~[FlameCord.jar:git:Travertine-Bootstrap:1.16-R0.4-SNAPSHOT:4d67d99:unknown]
        at net.md_5.bungee.BungeeCord.start(BungeeCord.java:319) ~[FlameCord.jar:git:Travertine-Bootstrap:1.16-R0.4-SNAPSHOT:4d67d99:unknown]
        at net.md_5.bungee.BungeeCordLauncher.main(BungeeCordLauncher.java:49) ~[FlameCord.jar:git:Travertine-Bootstrap:1.16-R0.4-SNAPSHOT:4d67d99:unknown]
        at net.md_5.bungee.Bootstrap.main(Bootstrap.java:15) ~[FlameCord.jar:git:Travertine-Bootstrap:1.16-R0.4-SNAPSHOT:4d67d99:unknown]
Caused by: java.sql.SQLException: Malformed communication packet
        at me.lucko.luckperms.lib.mysql.jdbc.SQLError.createSQLException(SQLError.java:965) ~[?:?]
        at me.lucko.luckperms.lib.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3933) ~[?:?]
        at me.lucko.luckperms.lib.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3869) ~[?:?]
        at me.lucko.luckperms.lib.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2524) ~[?:?]
        at me.lucko.luckperms.lib.mysql.jdbc.ServerPreparedStatement.serverExecute(ServerPreparedStatement.java:1281) ~[?:?]
        at me.lucko.luckperms.lib.mysql.jdbc.ServerPreparedStatement.executeInternal(ServerPreparedStatement.java:787) ~[?:?]
        at me.lucko.luckperms.lib.mysql.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:2020) ~[?:?]
        at me.lucko.luckperms.lib.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52) ~[?:?]
        at me.lucko.luckperms.lib.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java) ~[?:?]
        at me.lucko.luckperms.common.messaging.sql.AbstractSqlMessenger.init(AbstractSqlMessenger.java:79) ~[?:?]
        at me.lucko.luckperms.common.messaging.sql.SqlMessenger.init(SqlMessenger.java:55) ~[?:?]
        ... 11 more
[17:41:19 WARN] [LuckPerms]: Messaging service 'sql' not recognised.
[17:41:19 INFO] [LuckPerms]: Loading internal permission managers...
[17:41:19 INFO] [LuckPerms]: Performing initial data load...
[17:41:19 ERROR]: java.util.concurrent.CompletionException: java.sql.SQLException: Malformed communication packet
[17:41:19 ERROR]:       at me.lucko.luckperms.common.storage.Storage.lambda$makeFuture$1(Storage.java:105)
[17:41:19 ERROR]:       at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
[17:41:19 ERROR]:       at net.md_5.bungee.scheduler.BungeeTask.run(BungeeTask.java:66)
[17:41:19 ERROR]:       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[17:41:19 ERROR]:       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[17:41:19 ERROR]:       at java.lang.Thread.run(Thread.java:748)
[17:41:19 ERROR]: Caused by: java.sql.SQLException: Malformed communication packet
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.mysql.jdbc.SQLError.createSQLException(SQLError.java:965)
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3933)
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3869)
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2524)
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.mysql.jdbc.ServerPreparedStatement.serverExecute(ServerPreparedStatement.java:1281)
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.mysql.jdbc.ServerPreparedStatement.executeInternal(ServerPreparedStatement.java:787)
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.mysql.jdbc.PreparedStatement.executeQuery(PreparedStatement.java:2020)
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.hikari.pool.ProxyPreparedStatement.executeQuery(ProxyPreparedStatement.java:52)
[17:41:19 ERROR]:       at me.lucko.luckperms.lib.hikari.pool.HikariProxyPreparedStatement.executeQuery(HikariProxyPreparedStatement.java)
[17:41:19 ERROR]:       at me.lucko.luckperms.common.storage.implementation.sql.SqlStorage.selectGroups(SqlStorage.java:923)
[17:41:19 ERROR]:       at me.lucko.luckperms.common.storage.implementation.sql.SqlStorage.loadAllGroups(SqlStorage.java:485)
[17:41:19 ERROR]:       at me.lucko.luckperms.common.storage.Storage.lambda$loadAllGroups$10(Storage.java:194)
[17:41:19 ERROR]:       at me.lucko.luckperms.common.storage.Storage.lambda$makeFuture$1(Storage.java:100)
[17:41:19 ERROR]:       ... 5 more```

**Comment by Fefo6644** [comment]
What is the `storage-method` in your LP config?

**Comment by Sergix7720** [comment]
mysql

**Comment by Fefo6644** [comment]
Set it to `mariadb` then. Should fix the issue, see #2701

**Comment by Sergix7720** [comment]
Ok, it's fixed, thanks!
